xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/html/DefaultHTMLCleaner.java

```diff
@@ -222,6 +222,9 @@ private CleanerProperties getDefaultCleanerProperties(HTMLCleanerConfiguration c
         defaultProperties.setHtmlVersion(4);
 
         // Don't trim spaces in attribute values
+        // We don't want input such as: <input type="hidden" value="  foo" />
+        // to be modified for: value="foo"
+        // if such case happen, the value should remain the same.
```

Comment:
Thanks but you still don't explain why. The comment you put is not needed since that's what 'setTrimmedAttributeValues(false)` is about. You're just explaining WHAT the code does but not WHY :)

I was asking to explain why we want `value = "  foo"` instead of `value = "foo"` since it's far from obvious and it's not the default!

Reply:
Sorry but I still don't see why we wouldn't want spaces to be trimmed in attribute values. Could you explain? I guess that if there was a JIRA issue then there's a reason why this JIRA issue was created.

Reply:
Also, if there's a good reason to not trim spaces, we might want to raise an issue for htmlcleaner to ask them to change the default and explain why it's not a good default. Unless the reason is specific to XWiki ofc.

Reply:
> So again, IMO it should only concern form value attribute. So maybe I should create a filter to only clean those "value" attribute.

Ok I understand now (even though I don't know the exact use case, I can understand that we'd want , in some cases, to keep exactly what the use has entered in a form).

Now the change above removes the trim for all attributes. I wonder if this can cause some troubles in some part of our code where we had leading or trailing spaces that were fixed by the html cleaner. Also if users use our `{{html}}` macro the HTML Cleaner is executed and this if they were using HTML with attribute values with leading/trailing spaces maybe it was good that our HTML cleaner was "fixing" them for their use cases and maybe their use case will fail after the change?

So indeed, it could be safer to have a filter for that to limit the trimming for form elements.

Now maybe this is overkill, I'm not sure.

In any case, for the comment, I'd say the we need HTML Form elements to not have their attributes trimmed because we want to keep the user-entered content intact (as it was entered) when doing further processing on the values.

Reply:
> Now maybe this is overkill, I'm not sure.

The only usecase I can see where we might break something, is for example in case of an html containing classes with leading space and a javascript checking exactly the class attribute value to do something. 

So it's clearly an edge usecase, but it might exist indeed. 

> So indeed, it could be safer to have a filter for that to limit the trimming for form elements.

I'll reopen the issue then.